src/fetchtastic/download/async_client.py

```
@@ -0,0 +1,553 @@
+
+            try:
+                start_time = time.time()
+
+                async with session.get(url) as response:
+                    if response.status >= 400:
+                        raise AsyncDownloadError(
+                            f"HTTP error {response.status}",
+                            url=url,
+                            status_code=response.status,
+                            is_retryable=response.status >= 500,
+                        )
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)